### DISCUSSION TITLE
How do I get my model to work correctly?

### DISCUSSION THREAD

**Discussion by mengysu** [google-deepmind/mujoco#260] [Asking for Help] [answered]
Dear MuJoCo developers,
I am trying to use my humanoid model (21 freedom) for reinforcement learning, based on mujoco and mujoco-py. So far, I encountered some issues:
1. In my XML file, when I set the property `armature = 0.01,` It works fine if I just give the command, but if I train my model, it can not work well. When I set `armature = 2.5`, the model falls back if I give a direct command, but I can train for some bad effects with my model, like falling over after a single step. That confuses me a lot, what does ‘armature’ mean and how should I set the value? 
2. In the process of learning the model, the agent interacts with the environment through `step(action)`, where the code that interacts with the environment is `do_simulation(ctrl, skip_frame)`. No matter if I set  `motor` or `position` in the `<actuator>`, the joint only can move to a certain position, it seems like controlling the position. The difference is that the latter one is very shaky. I set the `<actuator> `as follows:
```
	<actuator>
		<motor name="waist"  gear="100.00000" joint="waist "/>
		<motor name="left_arm"   gear="100.00000" joint="left_arm"/>
		<motor name="right_arm"  gear="100.00000" joint="right_arm"/>
		<motor name="left_leg_1" gear="100.00000" joint="left_leg_1"/>
		<motor name="left_leg_2" gear="100.00000" joint="left_leg_2"/>
		<motor name="left_leg_3" gear="100.00000" joint="left_leg_3"/>
		<motor name="left_leg_4" gear="100.00000" joint="left_leg_4"/>
		<motor name="left_leg_5" gear="100.00000" joint="left_leg_5"/>
		<motor name="left_leg_6" gear="100.00000" joint="left_leg_6"/>
		<motor name="right_leg_1" gear="100.00000" joint="right_leg_1"/>
		<motor name="right_leg_2" gear="100.00000" joint="right_leg_2"/>
		<motor name="right_leg_3" gear="100.00000" joint="right_leg_3"/>
		<motor name="right_leg_4" gear="100.00000" joint="right_leg_4"/>
		<motor name="right_leg_5" gear="100.00000" joint="right_leg_5"/>
		<motor name="right_leg_6" gear="100.00000" joint="right_leg_6"/>
	</actuator>
```

I found that when I directly gave joint angles, it required a transformation, I got the relationship  `*100/7` through trial. I want to know why both motor and position are position control, as well as the actual process of control and their true transformation relationship.

**Comment by yuvaltassa** [comment, marked as answer]
1. Re armature inertia, see [here](https://mujoco.readthedocs.io/en/latest/XMLreference.html?highlight=armature#body-joint).
2. Re position actuators and length scaling see #249

**Reply by mengysu** [reply]
Thanks for your answer, but I also have some problems:
1. I have read the document but I still wonder why different armature has such different results, and how should I choose the correct value?
2. Position actuator means I should give` radin/100`, and motor actuator means I should give `radin*100`? However, the actual output is not correct, if I give the value `radin*100/7` (maybe), the model will work well.

**Reply by yuvaltassa** [reply]
1. Try different values of armature to understand what it does. Maybe maybe make a test model with a single joint? I don't know how you should choose since I don't know what "correct" means for you. It depends on what you are trying to achieve.
2. Don't use `gear` for position actuators at all, use `kp`. Then the semantics will be what you expect.
